Task05/src/main.cpp

```diff
@@ -0,0 +1,227 @@
+#include <iostream>
+#include <fstream>
+#include <unordered_map>
+#include <string>
+#include <sstream>
+#include <cmath>
+#include <functional>
+#include <algorithm>
+#include <vector>
+#include <array>
+#include "frequencies.h"
+
+
+
+
+std::string filename = "data.txt";
+
+/**
+ * Calculates the Euclidean distance between two points (x1, y1) and (x2, y2).
+ */
+double calculateDistanceL2(double x1, double y1, double x2, double y2) {
+    return std::sqrt(std::pow(x1 - x2, 2) + std::pow(y1 - y2, 2));
+}
+
+/**
+ * Calculates the Chebyshev distance between two points (x1, y1) and (x2, y2).
+ */
+double calculateDistanceLinf(double x1, double y1, double x2, double y2) {
```

Comment:
there is std::function in the STL library that you can use and not a pointer to function